frontend/leaflet-map/map.js

```diff
@@ -0,0 +1,128 @@
+var map = L.map('map').setView([37.349167, -121.938056], 16);
+
+L.tileLayer('http://a.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+  attribution: 'Map data &copy; OpenStreetMap contributors',
+  maxZoom: 18
+}).addTo(map);
+
+///////////////////////////////////////////////////
+// Utility to return needed colors for a range
+function getColor(d) {
+  return d > 500  ? '#ff0000' :
+         d > 200  ? '#ff4000' :
+         d > 100  ? '#ff8000' :
+         d > 50   ? '#ffbf00' :
+         d > 20   ? '#ffff00' :
+         d > 10   ? '#bfff00' :
+                    '#80ff00';
+}
+
+function getRandomInt(max) {
+  return Math.floor(Math.random() * max);
+}
+
+///////////////////////////////////////////////////
+// Custom Control for selecting day of week
+var daySelector = L.control();
+
+daySelector.onAdd = function (map) {
+  this._div = L.DomUtil.create('div', 'dow');
+  this._div.innerHTML = '<label for="dow">Day of Week: </label>' +
+  '<select id="dayOfWeek">' +
+  '<option value="1" selected="selected">Monday</option>' +
+  '<option value="2">Tuesday</option>'   +
+  '<option value="3">Wednesday</option>' +
+  '<option value="4">Thursday</option>'  +
+  '<option value="5">Friday</option>'    +
+  '<option value="6">Saturday</option>'  +
+  '<option value="7">Sunday</option>'    +
+  '</select><br>';
+
+  return this._div;
+};
+
+daySelector.onRemove = function(map) {
+  // L.DomEvent.off();
+}
+
+daySelector.addTo(map);
+
+///////////////////////////////////////////////////
+// Custom Info Control Displaying Stats
+var info = L.control();
+
+info.onAdd = function (map) {
+  this._div = L.DomUtil.create('div', 'info'); // create a div with a class "info"
+  this.update();
+  // L.DomEvent.on();
+  return this._div;
+};
+
+info.onRemove = function(map) {
+  // L.DomEvent.off();
+}
+
+// method that we will use to update the info based on map information passed in
+info.update = function (props) {
+  this._div.innerHTML = '<h4>Road Traffic Statistics</h4>';
+  if (!props)
+    this._div.innerHTML += 'Click a location';
+  else
+  {
+    var dow = ["MON", "TUE", "WED", "THR", "FRI", "SAT", "SUN"]; 
+
+    this._div.innerHTML += '<b>' + dow[props.value-1] + '(' + props.value +') traffic stats</b> @ ' 
+      + props.fix.lat.toFixed(3) + ' | ' + props.fix.lng.toFixed(3);
+
+    // <<<<<< TODO <<<<<<<
+    // Call BE-API to get {morning:mValue, afternoon:aValue, evening:eValue}
+    var resp = {morning:getRandomInt(500), afternoon:getRandomInt(500), evening:getRandomInt(500)};
```

Comment:
1) FE sends dayOfWeek (1-7) and gps fix (lat, lng) in a POST http://localhost:8080/mappapp?dow=1&lat=123.3434242&lng=341.323132


2) BE sends back JSON like this
{morning:$mValue, afternoon:$aValue, evening:$eValue}
mValue, aValue, eValue are positive number denotes how busy are the road.

Reply:
Correction: ;day of the week (0 to 6) Sunday = 0, Monday = 1